src/solver/conv_hip_implicit_gemm_fwd_v4r4_xdlops_nchw_kcyx_nkhw.cpp

```diff
@@ -0,0 +1,961 @@
+
+        // SrcDataPerRead_GemmN bounded by size of threadwise copy
+        SrcDataPerRead_GemmN = gcd(SrcDataPerRead_GemmN, data_per_thread_copy);
+
+        // SrcDataPerRead also bounded by GemmKPack;
+        SrcDataPerRead_GemmN = gcd(SrcDataPerRead_GemmN, GemmNPerBlock);
+
+        const auto data_per_thread_copy_gemmn = SrcDataPerRead_GemmN;
+        const auto tmp                        = data_per_thread_copy / data_per_thread_copy_gemmn;
+
+        int data_per_thread_copy_gemmkpack = 0;
+        int data_per_thread_copy_gemmk     = 0;
+
+        if(GemmBThreadCopyMoreGemmKPack)
+        {
+            data_per_thread_copy_gemmkpack = gcd(GemmKPack, tmp);
+            data_per_thread_copy_gemmk     = tmp / data_per_thread_copy_gemmkpack;
+        }
+        else
+        {
+            data_per_thread_copy_gemmk     = gcd(GemmKPerBlock, tmp);
+            data_per_thread_copy_gemmkpack = tmp / data_per_thread_copy_gemmk;
+        }
+
+        // vector write into LDS
+        DstDataPerWrite_GemmKPack = gcd(DstDataPerWrite_GemmKPack, data_per_thread_copy_gemmkpack);
+
+        if(!(GemmKPerBlock % data_per_thread_copy_gemmk == 0 &&
+             GemmNPerBlock % data_per_thread_copy_gemmn == 0 &&
+             GemmKPack % data_per_thread_copy_gemmkpack == 0))
+            MIOPEN_THROW("invalid performance parameter");
+
+        ClusterLengths_GemmK     = GemmKPerBlock / data_per_thread_copy_gemmk;
+        ClusterLengths_GemmN     = GemmNPerBlock / data_per_thread_copy_gemmn;
+        ClusterLengths_GemmKPack = GemmKPack / data_per_thread_copy_gemmkpack;
+    }
+    catch(...)
+    {
+        return std::make_tuple(-1, -1, -1, -1, -1, false);
+    }
+
+    return std::make_tuple(ClusterLengths_GemmK,
+                           ClusterLengths_GemmN,
+                           ClusterLengths_GemmKPack,
+                           SrcDataPerRead_GemmN,
+                           DstDataPerWrite_GemmKPack,
+                           true);
+}
+
+std::tuple<std::size_t, bool> PerformanceImplicitGemmForwardV4R4Xdlops::CalculateLdsNumberOfByte(
+    const ConvolutionContext& ctx) const
+{
+    const auto a_block_space = GemmKPerBlock * GemmMPerBlock * GemmKPack;
+    const auto b_block_space = GemmKPerBlock * GemmNPerBlock * GemmKPack;
+
+    std::size_t lds_size =
+        (a_block_space + b_block_space) * (ctx.IsFp32() ? sizeof(float) : sizeof(half_float::half));
+
+    return std::make_tuple(lds_size, true);
+}
+
+// Return false, if you don't want to this to be included in tuning range used by generic search
+// Should only be used for tuning
+// A performance config may still be valid w.r.t algorithm correctness, even when IsValid() return
+// false
+bool PerformanceImplicitGemmForwardV4R4Xdlops::IsValid(const ConvolutionContext& ctx) const
+{
+    return IsValidValue() && IsValidPerformanceConfig(ctx) &&
+           (!IsWorkAroundedPerformanceConfig(ctx)) && IsFastPerformanceConfig(ctx);
```

Comment:
Please remove extra pair of parentheses.

Reply:
removed